yahoo/kafka-manager
# API: Clusters and TopicIdentity end points.

#### What's this PR do?

This PR adds following API end points:
* `/api/status/clusters` => Returns a list of active and pending clusters and their metadata
* `/api/status/:c/topicIdentities` => Returns a list of topic identities for a given cluster

#### Where should the reviewer start?

* `app/controllers/api/KafkaStateCheck.scala/clusters()`
* `app/controllers/api/KafkaStateCheck.scala/topicIdentities()`

#### How should this be manually tested?

1. Go to `your-kafka-manager.com/api/status/clusters` in your favourite browser.
2. Verify the list of clusters are displayed in valid JSON.

#### Any background context you want to provide?

This work was implemented so that we can monitor various different metrics in our kafka clusters. The credit goes to @kelseylam and @khhirani.
